apps/web/src/server/applications/pages/case/applications-case.locals.js

```diff
@@ -0,0 +1,62 @@
+import { url } from '../../../lib/nunjucks-filters/index.js';
+import { getCase } from '../../lib/services/case.service.js';
+import {
+	getCaseDocumentationFolderPath,
+	getCaseFolder
+} from './documentation/applications-documentation.service.js';
+
+/**
+ * @typedef {object} ApplicationCaseLocals
+ * @property {number} caseId
+ * @property {number} folderId
+ * @property {import('../../applications.types').Case} Case
+ */
+
+/**
+ * @typedef {object} FolderLocals
+ * @property {import('../../applications.types').DocumentationCategory} currentFolder
+ * @property {Array<{href: string, text: string}>} breadcrumbItems
+ */
+
+/**
+ * Use the `caseId` parameter to install the Case onto the request
+ * locals, for consumption in subsequent guards and controllers.
+ *
+ * @type {import('@pins/express').RequestHandler<ApplicationCaseLocals>}
+ */
+export const registerCase = async (request, response, next) => {
+	response.locals.caseId = Number(request.params.caseId);
+	response.locals.case = await getCase(response.locals.caseId);
+
+	next();
+};
+
+/**
+ * Register the current folder and the items (url and text) for the breadcrumbs component
+ *
+ * @type {import('@pins/express').RequestHandler<ApplicationCaseLocals>}
+ */
+export const registerFolder = async ({ params }, response, next) => {
+	const folderId = Number.parseInt(params.folderId, 10);
+	const { caseId } = response.locals;
+
+	response.locals.folderId = folderId;
+
+	// get the current folder
+	response.locals.currentFolder = await getCaseFolder(caseId, folderId);
+
+	// get the folderTree for breadcrumbs
```

Comment:
@DavidJWhiteDev , this is the shared functions I was talking about. They register folderId, currentFolder and folderTree for every page in the documentation domain.

Also, I moved some logic from the template to the js. easier to read than the mysterious nunjucks syntax.